cyw43/src/hci_connector.rs

```diff
@@ -0,0 +1,36 @@
+pub struct HciConnector {}
+
+impl HciConnector {
+    pub fn new() -> HciConnector {
+        return HciConnector {};
+    }
+}
+
+#[derive(Debug)]
+pub enum HciConnectorError {
+    Unknown,
+}
+
+impl embedded_io_async::Error for HciConnectorError {
+    fn kind(&self) -> embedded_io_async::ErrorKind {
+        embedded_io_async::ErrorKind::Other
+    }
+}
+
+impl embedded_io_async::ErrorType for HciConnector {
+    type Error = HciConnectorError;
+}
+
+impl embedded_io_async::Read for HciConnector {
+    async fn read(&mut self, buf: &mut [u8]) -> Result<usize, HciConnectorError> {
+        // TODO: how to get all the way to runner.hci_read()?
+        Ok(0)
+    }
+}
+
+impl embedded_io_async::Write for HciConnector {
+    async fn write(&mut self, buf: &[u8]) -> Result<usize, HciConnectorError> {
+        // TODO: how to get all the way to runner.hci_write()?
```

Comment:
I don't know anything about HCI but the general idea would probably be to turn the select3 in the runner into select4 with another channel for bluetooth output. So bytes that are to be written here are pushed into a channel. The runner is then awoken on this channel and do the `hci_write`.

For reads, I assume it starts with an interrupt? After an interrupt, you check why you were interrupted. Previously there was basically only wifi data available. So then you would need another channel/buffer there that you push the data into. The other side would then be woken up and can consume the bytes.